main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -578,4 +578,6 @@ object Ast {
       */
     case class FallbackText(s: String) extends VarText
   }
+
+  case class PurityAndEffect(pur: Option[Type], eff: Option[List[Type]])
```

Comment:
Whats the intuition for Option List? That its always a union?

Reply:
We have to distinguish between the empty set and the absent effect.
`& ef \ { }` means that `ef` is a bool, wheras `& ef` (without the effect part) is both effect and bool.